### DISCUSSION TITLE
Import order rule

### DISCUSSION THREAD

**Discussion by theprobugmaker** [eslint/eslint#13976] [Q&A] [answered]
Is there a rule (or how to create it) that won't allow local imports to be mixed with library imports? It would require an empty line between them, for example:

wrong:
```
import foo from 'foo'
import bar from 'bar'
import biz from './biz'
import boz from './boz'
```

wrong:
```
import biz from './biz'
import foo from 'foo'
import bar from 'bar'
import boz from './boz'
```

correct:
```
import foo from 'foo'
import bar from 'bar'

import biz from './biz'
import boz from './boz'
```

Thank you

**Comment by mdjermanovic** [comment, marked as answer]
Hi @zefexdeveloper!

I think the rule you're looking for is [import/order](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/order.md) with option [newlines-between](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/order.md#newlines-between-ignorealwaysalways-and-inside-groupsnever) set to `"always"`.

**Comment by theprobugmaker** [comment]
@mdjermanovic Thank you for the answer and I'm sorry the delay answering, I was traveling.
I installed the `import/order` plugin and set the rule to `always` for `newlines-between` but it's adding the lines in between the local imports.

Example:
```
import foo from 'foo'
import bar from 'bar'

import biz from './biz'

import boz from './boz'
```

I want the space only between the local and 3rd libraries.

**Reply by mdjermanovic** [reply]
> I installed the `import/order` plugin and set the rule to `always` for `newlines-between` but it's adding the lines in between the local imports.
> 
> Example:
> 
> ```
> import foo from 'foo'
> import bar from 'bar'
> 
> import biz from './biz'
> 
> import boz from './boz'
> ```

It could be a bug in the rule. I couldn't reproduce it, though.

Since this is a third-party rule and we are not maintaining it in this repository, can you please open an issue in `eslint-plugin-import` repo:

https://github.com/benmosher/eslint-plugin-import

**Comment by theprobugmaker** [comment]
Thank you, I did and they answered there: https://github.com/benmosher/eslint-plugin-import/issues/1971
